Finished add_date_to_folders feature

eggera opened the pull request:
**New config**

The config option `add_date_to_folders` will create a date folder as a subfolder from the selected config (i.e. ga/2015-05-06). 

Inside the date folder the generated files will be saved prefixed with a timestamp (i.e. 114430 for 11:44:30) and an indicator whether the files are input or output (input = requests,servers; output = results,schedule)

Sample files would be 
- 114430_input_requests
- 114430_input_servers
- 114430_output_results
- 114430_output_schedule

Saved in folder ga/2015-05-06/ (or whatever the current date is). 

Now the inputs as well as the outputs are saved at a common place easily to be recognized by a unique timestamp. If you don't need this option, simply deactivate it by setting `add_date_to_folders = False` in your custom settings file.

Commit:
Amending number of servers and VMs

Commit:
Reset base.py, add new simple configuration

Commit:
Adjust output to date folders with file timestamps

Commit:
Adding ipython debugger option to gascheduler

Commit:
Documenting format_spec method

Commit:
Amend simple configuration file

Commit:
Rearrange output in folders + date, configurable

Commit:
Finished "add_date_to_folders" feature

Commit:
Quick bugfix when setting power_freq_model to Fals

Commit:
Switching to No temperature setting working

Commit:
Adding simple scheduler and environment

metakermit commented:
Thank you, Andreas. There were many things in there, so I edited some parts out (like the simple scheduler) and squashed it into one commit. If you send more future commits, try to make them atomic and small like that :)

This is what it looks like in the main repo:

https://github.com/philharmonic/philharmonic/commit/c5132c5cab043f37879ab4d5808fd3b623d39687

I rewrote your `loc(conf.cloud_input... + filepath, input=True)` to `input_loc(filepath)`. Nicer separation of concerns that way. Also `input` is a built-in function in Python, so you shouldn't name your variables like that. I renamed it to `input_loc` when used as a parameter to your date-based function. Other than that, it works quite nicely with the time-stamps. Might just use it myself now :)

Also, I don't know if disabling the temperature will work now - I had to change it in a couple of places, as it was throwing errors on me.

eggera commented:
Hi Drazen, 

Ok thanks, it works now. There just seems to be a problem with the forecast_window_index function in environment.py:

https://github.com/philharmonic/philharmonic/blob/master/philharmonic/simulator/environment.py#L150

When called from the brute_force_scheduler it raises an error from pandas tseries, raise ValueError ("Could not evaluate %s" % freqstr)  - freqstr is 1:00:00 at that time

So for now I excluded the bruteforcescheduler tests with @nottest annotations, then all tests run through. 

Concerning the add_date_to_folders feature,  I meant to separate the files generated for input and for output, so I made 2 functions to get the in/output folders which is input_loc and output_loc for the files that should be inputs and outputs respectively. I removed the loc function as it would not be needed anymore. I hope this does not interfere too much, but I wanted to finish it as soon as possible. 

After a second thought the output of the simulator should for me ideally be in csv format, such that I can further process it and make graphs etc. I know this can probably be done also with python libraries, but I think for me it would be easier that way. I thought about sth like “date,price,temperature.(if available),utilization,nrVMs,runningServers” for each location, s.t. I can see which energy price changes caused VMs to be migrated to other DCs. I will see how this would be possible. There would be an option to generate csv files or not. 

Also I re-read part of the paper you recommended me again [1] and found a great reference to a paper that does very similar investigations than what I had in mind (cost aware assignments, but no migrations) [2]. 

Still trying to figure out what would make the thesis most interesting and what should I actually show. 

Best, 

Andreas

[1] De Cauwer, M., & O’Sullivan, B. (2013). A Study of Electricity Price Features on Distributed Internet Data Centers. In Economics of Grids, Clouds, Systems, and Services (pp. 60-73). Springer International Publishing.

[2] Rao, L., Liu, X., Xie, L., & Liu, W. (2010, March). Minimizing electricity cost: optimization of distributed internet data centers in a multi-electricity-market environment. In INFOCOM, 2010 Proceedings IEEE (pp. 1-9). IEEE.

Von: Dražen Lučanin [mailto:notifications@github.com] 
Gesendet: Freitag, 15. Mai 2015 16:59
An: philharmonic/philharmonic
Cc: and2egg
Betreff: Re: [philharmonic] Finished add_date_to_folders feature (#1)

Thank you, Andreas. There were many things in there, so I edited some parts out (like the simple scheduler) and squashed it into one commit. If you send more future commits, try to make them atomic and small like that :)

This is what it looks like in the main repo:

 https://github.com/philharmonic/philharmonic/commit/c5132c5cab043f37879ab4d5808fd3b623d39687 c5132c5

I rewrote your loc(conf.cloud_input... + filepath, input=True) to input_loc(filepath). Nicer separation of concerns that way. Also input is a built-in function in Python, so you shouldn't name your variables like that. I renamed it to input_loc when used as a parameter to your date-based function. Other than that, it works quite nicely with the time-stamps. Might just use it myself now :)

Also, I don't know if disabling the temperature will work now - I had to change it in a couple of places, as it was throwing errors on me.

—
Reply to this email directly or  https://github.com/philharmonic/philharmonic/pull/1#issuecomment-102423172 view it on GitHub.  https://github.com/notifications/beacon/ABeP5ewnENK1gY4qfjJBtekyhA0OQagGks5oJgFFgaJpZM4EaRuf.gif

metakermit commented:
Yes, we were also having some problems with those brute_force tests and the method is incomplete anyway, so I think we commented out those tests in the end. I see now that we haven't pushed those changes yet, but it will happen tomorrow probably.

Yeah, having CSV output would be a nice feature, so feel free to submit another pull request if you end up implementing this (this time in a nice little isolated commit that only includes that feature would be great ;) ). Cheers!

and2egg commented:
Hi Drazen!

 

Wow it has been long since the last email, sorry I have been a bit sick and also searching for jobs, so this got me off the plan a bit. 

Most importantly I have to find a job now, and also prepare for that as I will end my work at Siemens by end of September. 

Anyways I still did some things in the meantime, but Java this time, not python. I set up an application server in eclipse, installed a full featured oracle database and already tried out some samples for Java EE applications. It worked well, and I already have a working scheduler, that executes tasks periodically. It just doesn’t do anything meaningful yet ;-)

 

So actually I am way off my plan, almost impossible to do how I planned it. I will continue as much as I can, but I’m afraid there won’t be so much time available. Especially since the implementation will be still quite a big part. Anyways I’m committed to give my best, and to make it work. 

 

I also read a few papers again lately, and did some notes and summaries for the thesis, which I think might be useful. Will need to get a bit into theory again to be able to sum up and describe what I want to do. 

 

I will see what I can do in the simulator, but I want to keep implementation at a minimum. Otherwise I don’t think I will ever finish. 

 

Thanks and cu soon!

 

Andreas

 

 

Von: Dražen Lučanin [mailto:notifications@github.com] 
Gesendet: Sonntag, 17. Mai 2015 22:01
An: philharmonic/philharmonic
Cc: and2egg
Betreff: Re: [philharmonic] Finished add_date_to_folders feature (#1)

 

Yes, we were also having some problems with those brute_force tests and the method is incomplete anyway, so I think we commented out those tests in the end. I see now that we haven't pushed those changes yet, but it will happen tomorrow probably.

Yeah, having CSV output would be a nice feature, so feel free to submit another pull request if you end up implementing this (this time in a nice little isolated commit that only includes that feature would be great ;) ). Cheers!

—
Reply to this email directly or view it on GitHub <https://github.com/philharmonic/philharmonic/pull/1#issuecomment-102844470> .  <https://github.com/notifications/beacon/ABeP5bSFlu7eGDrSEG6rd8iNjXpsFHl_ks5oKOsUgaJpZM4EaRuf.gif>

metakermit commented:
It's better to have these types of conversations via e-mail and limit the talk on GitHub to technical issues only (related to the simulator).
